### PR TITLE
feat: Add device group hierarchy support

### PR DESCRIPTION
**Pull request by shinmog**

### CODE REVIEW

File: panos/panorama.py [hunk 1]
```diff
@@ -751,6 +789,34 @@ def get_vm_auth_keys(self):
 
         return ans
 
+    def get_device_group_hierarchy(self):
```
Comment (btorresgil) [inline panos/panorama.py:792]: The method's name implies getting a hierarchy, which to me implies a nested dict. But as you say in the docstring it's a flat dict where the value is the DG parent. Perhaps another name could better represent this expectation?

File: panos/panorama.py [hunk 2]
```diff
@@ -91,6 +91,44 @@ def devicegroup(self):
...
+        em = ET.SubElement(e, "move-dg")
+        eme = ET.SubElement(em, "entry")
+        eme.attrib["name"] = self.name
+
+        if parent is not None:
+            ET.SubElement(eme, "new-parent-dg").text = parent
+
+        cmd = ET.tostring(e, encoding="utf-8")
+        resp = dev.op(cmd, cmd_xml=False)
+        return dev.syncjob(resp)
```
Comment (btorresgil) [inline panos/panorama.py:130]: I don't think we should force an async operation to act synchronously. Perhaps a `sync=False` argument would be better to make this sync optional? Or let the developer call `syncjob()` themselves on the result of this method.

**Review by btorresgil — CHANGES_REQUESTED** [review]

**Comment by shinmog** [comment]
sigh, commit history is messed up, going to redo this

**Comment by tom0010** [comment]
Hey @shinmog did you do another PR for this? I just ran into this issue, so it'd be great if it could be merged :)

**Pull request closed without merging**